src/components/FileDropZone/FileDropZone.Provider.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+
+import {useActionHandlers, useDropZone, useFileInput} from '../../hooks';
+import type {UseDropZoneState, UseFileInputResult} from '../../hooks';
+
+import type {FileDropZoneProps} from './FileDropZone';
+import {cnFileDropZone} from './FileDropZone.classname';
+
+interface FileDropZoneContextValue
+    extends Pick<
+            FileDropZoneProps,
+            | 'title'
+            | 'description'
+            | 'buttonText'
+            | 'multiple'
+            | 'icon'
+            | 'errorIcon'
+            | 'errorMessage'
+            | 'validationState'
+        >,
+        UseFileInputResult,
+        UseDropZoneState {}
+
+const FileDropZoneContext = React.createContext<FileDropZoneContextValue | null>(null);
+
+export const FileDropZoneProvider = ({
```

Comment:
We usually give `Provider` postfix to entities that don't have a layout. For example: [TabProvider](https://github.com/gravity-ui/uikit/blob/main/src/components/tabs/TabProvider.tsx#L10), [ToasterProvider](https://github.com/gravity-ui/uikit/blob/d716102ed6138ea1665739fa973234820067d517/src/components/Toaster/Provider/ToasterProvider.tsx#L15), [AccordionProvider](https://github.com/gravity-ui/uikit/blob/d716102ed6138ea1665739fa973234820067d517/src/components/Accordion/AccordionContext.tsx#L53C17-L53C34), [DisclosureProvider](https://github.com/gravity-ui/uikit/blob/d716102ed6138ea1665739fa973234820067d517/src/components/Disclosure/DisclosureContext.tsx#L33C17-L33C35) etc